xdsl/tools/riscv-parser

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import sys
+from io import StringIO
+
+from xdsl.ir import MLContext
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.printer import Printer
+
+from xdsl.dialects.riscv import RISCV
+from xdsl.tools.riscv.parser import Parser as RVParser
+from xdsl.tools.riscv.printer import print_program
+
+from typing import Dict, Callable, List
+
+
+class Main:
```

Comment:
Is this a useful tool to add to the xDSL project? Seems like a yes, probably, but I'm not 100% sure

Reply:
I think so, but that was the main reason to open this PR as a draft. I was waiting for the new parser to land to factorize this into it. I think it would make more sense to have this in the main tool.